backend/src/BuiltinExecution/Libs/String.fs

```diff
@@ -304,12 +304,16 @@ let fns : List<BuiltInFn> =
       fn =
         (function
         | _, _, [ DString s; DInt first; DInt last ] ->
-          let strLength = System.Globalization.StringInfo(s).LengthInTextElements
+          let getLengthInTextElements s =
+            System.Globalization.StringInfo(s).LengthInTextElements
+
           // Handle negative indexes (which allow counting from the end)
-          let first = if first < 0 then strLength + int first else int first
-          let last = if last < 0 then strLength + int last else int last
+          let first =
+            if first < 0 then getLengthInTextElements (s) + int first else int first
+          let last =
+            if last < 0 then getLengthInTextElements (s) + int last else int last
 
-          if first >= strLength || first >= last then
+          if first >= getLengthInTextElements (s) || first >= last then
```

Comment:
I hadn't noticed before but this calculates string length every time. Is this case necessary (that is, if we remove it does the `else` case come up with the same answer)? If not, better to remove it so we aren't working out the length when it's not necessary.